### PR TITLE
Implement `RotatedMinimumWidthBoundingRectangle`

### PR DESCRIPTION
**Pull request by peterstace**
## Description

This builds on the functionality implemented as part of the `RotatedMinimumAreaBoundingRectangle` function (see https://github.com/peterstace/simplefeatures/pull/582). It's a variant that searches for the rotated bounding rectangle that has the smallest width.

## Check List

Have you:

- Added unit tests? Yes.

- Add cmprefimpl tests? (if appropriate?) N/A

- Updated release notes? (if appropriate?) Yes.

## Related Issue

- N/A

### CODE REVIEW

File: geom/alg_rotating_calipers.go [hunk 1]
```diff
@@ -78,17 +97,18 @@ func findBestMBR(seq Sequence) rotatedRectangle {
...
 
-		area := rhs.proj.Sub(lhs.proj).Cross(far.proj)
-		if area < bestArea {
-			bestArea = area
-			rect = rotatedRectangle{
-				origin: seq.GetXY(i).Add(lhs.proj),
-				span1:  rhs.proj.Sub(lhs.proj),
-				span2:  far.proj,
-			}
+		canadidateRect := rotatedRectangle{
```
Comment (albertteoh) [inline geom/alg_rotating_calipers.go:100]: typo: canadidateRect -> candidateRect

File: geom/alg_rotating_calipers.go [hunk 2]
```diff
@@ -78,17 +97,18 @@ func findBestMBR(seq Sequence) rotatedRectangle {
...
+			origin: seq.GetXY(i).Add(lhs.proj),
+			span1:  rhs.proj.Sub(lhs.proj),
+			span2:  far.proj,
+		}
+		candidateMetric := metric(canadidateRect)
+		if i == 0 || candidateMetric < bestMetric {
```
Comment (albertteoh) [inline geom/alg_rotating_calipers.go:106]: It looks like we're getting the minimum metric, what do you think about renaming `best` -> `min` (e.g. `findMinMBR`, etc.) to be more explicit?

File: geom/alg_rotating_calipers.go [hunk 3]
```diff
@@ -8,25 +8,36 @@ import (
...
-// returned.
+// (zero area), then a point or line string (with a single line segment) will
+// be returned.
 func RotatedMinimumAreaBoundingRectangle(g Geometry) Geometry {
+	return rotatedMinimumBoundingRectangle(g, rotatedRectangle.area)
+}
+
+// RotatedMinimumWidthBoundingRectangle finds a rectangle with minimum width
+// that fully encloses the geometry. If the geometry is empty, the empty
+// geometry of the same type is returned. If the bounding rectangle would be
```
Comment (albertteoh) [inline geom/alg_rotating_calipers.go:19]: `s/would be/is/g`?

**Review by albertteoh — APPROVED** [review]
LGTM!

**Comment by peterstace** [comment]
Thanks for reviewing @albertteoh! I've implemented your suggestions in 7ec9ce9013097d11692e670dd373446c263690fc.

**Pull request merged**